### ISSUE TITLE
Leaving webpage leaves pending "connect" behind

### ISSUE THREAD

**Issue by sbordet** [cometd/cometd#388]
Latest dojo 1.8.0 improved OnLoad functionality (with dojo/ready and dojo.domReady!) to detect when DOM is ready and scripts can/should start to execute.

When user loads page that has comed app (including Reload extension) on it all is loaded properly and works fine. Problem happens when user leaves this page and loads another (or same page) which has same cometd app on it.

Attached image:
Image taken using latest FireFox. You can see calls that happen on any page which was loaded after page that has cometd on it. Particularly on this image page is empty and has no cometd app/code in it (this can not be observed for example in Chrome). Unfortunately, FireFox "sees" that last pending connect call (takes 20 seconds) as DOM loading call, so dojo/ready is waiting for it to finish loading and that disallows execution of dojo scripts on that page.

"Funny" is that:
- one can see calls from the image even if new page loaded has nothing related to cometd on it (plain page for example), so it must be related to previous page (firefox bug no closing previous pending calls?)
- other browsers do not have problem with this or they properly discard previous page opened/pending calls

So, either firefox has some implementation issues or cometd needs properly "close" opened connection on page unload, or allow us (developers) to call some sort of function which would not send disconnect to the server but will close opened connections on the client side (in the browser)

Or is this Reload extension "problem"? where it creates new connects on page unload which are incorrectly added in FireFox into new page stack?

Basically, calling cometd.reload() should tell cometd not to make any new "connects" past this point (it looks like FireFox "registers" that call into next page load instead of the leaving page, strange).

Yes it is most likely FireFox bug, but if cometd could prevent that last call, fixing it would be much fatser.
